File: Behavioral/Strategy/cpp/BankTransferPayment.cpp
#include "BankTransferPayment.h"
#include <iostream>

BankTransferPayment::BankTransferPayment(const std::string& accountNumber)
    : accountNumber(accountNumber) {
}

void BankTransferPayment::pay(double amount) {
    std::cout << "使用銀行轉帳支付 $" << amount << std::endl;
    std::cout << "帳號: " << accountNumber << std::endl;
    std::cout << "轉帳成功！" << std::endl;
} 

File: Behavioral/Strategy/java/BankTransferPayment.java
package com.designpatterns.strategy;

/**
 * 銀行轉帳支付策略
 * 具體策略實作：使用銀行轉帳進行支付
 */
public class BankTransferPayment implements PaymentStrategy {
    private final String accountNumber;

    public BankTransferPayment(String accountNumber) {
        this.accountNumber = accountNumber;
    }

    @Override
    public void pay(double amount) {
        System.out.println("使用銀行轉帳支付 $" + amount);
        System.out.println("帳號: " + accountNumber);
        System.out.println("轉帳成功！");
    }
} 